### PR TITLE
use std instead of boost timer

### PR DESCRIPTION
**Pull request by flynneva**
this should fix the ci build errors I think

---
This change is [Reviewable](https://reviewable.io/reviews/apexai/static_exception/1)

### CODE REVIEW

**Comment by flynneva** [comment]
also had to [update the GCC version in `src/exception_memory_pool.cpp` file line `24`](https://github.com/ApexAI/static_exception/blob/532835115cdb74c75c6731bcba3e3cfc186d50bd/src/exception_memory_pool.cpp#L24) to get it to compile on my workstation (run `gcc --version` to see your version) which had GCC v9.3.0.

let me know if you'd want to update that line as well in this PR?

**Comment by flynneva** [comment]
image

**Comment by flynneva** [comment]
realized late last night that we should probably also be printing out the cpu time as well using `std::clock`. added that in there this morning so now the test looks like this:

image

File: src/exception_memory_pool.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include "unwind-cxx.h"
 
 #ifdef __GNUC__
-#if __GNUC__==5 && __GNUC_MINOR__==4 && __GNUC_PATCHLEVEL__==0
+#if __GNUC__==9 && __GNUC_MINOR__==3 && __GNUC_PATCHLEVEL__==0
```
Comment (kylemarcey) [inline src/exception_memory_pool.cpp]: I would probably just make it check that the version is at least 5.4 so it doesn't have to be updated every time someone wants to build. I think these APIs should be reasonably stable

File: test/static_exception_test.cpp [hunk 2]
```diff
@@ -12,15 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <ctime>
```
Comment (kylemarcey) [inline test/static_exception_test.cpp:15]: You should include `chrono` as well

File: test/static_exception_test.cpp [hunk 3]
```diff
@@ -102,10 +100,12 @@ void check_used_segments(std::size_t expected) {
...
   check_used_segments(0);
-#ifdef ENABLE_BOOST_TIMER
-  boost::timer::cpu_timer t;
-  t.start();
-#endif
+
+  // use std::clock to measure cpu time
+  std::clock_t start = std::clock();
+  // use std::chrono to measure wall time
+  auto t_start = std::chrono::high_resolution_clock::now();
```
Comment (kylemarcey) [inline test/static_exception_test.cpp]: `steady_clock` should be used when measuring a duration since it's guaranteed to be monotonic

**Review by kylemarcey — CHANGES_REQUESTED** [review]

**Pull request merged**